contrib/postgresql/src/main/scala/doobie/contrib/postgresql/syntax.scala

```diff
@@ -8,7 +8,14 @@ import doobie.util.catchsql.exceptSomeSqlState
 /** Module of recovery combinators for PostgreSQL-specific SQL states. */
 object syntax {
 
-  implicit class SqlStateOps[M[_]: Monad: Catchable, A](ma: M[A]) {
+  implicit def toSqlStateOps[M[_]: Monad: Catchable, A](ma: M[A]) =
+    new SqlStateOps(ma)
+
+  implicit def toSqlStateOpsUnapply[FA](v: FA)
+    (implicit F0: Unapply[Monad, FA], F1: Unapply[Catchable, FA]) =
+    new SqlStateOps[F0.M, F0.A](F0(v))(F0.TC, F1.TC.asInstanceOf[Catchable[F0.M]])
```

Comment:
We know that `F0.M =:= F1.M` so this cast is fine, but it would be nice if we could prove it here. But we can't ask for implicit evidence here because the dependent method type rules don't allow it. I guess we may need another auxiliary type or something. Anyway, it works.